File: onyx-database-tests/src/entities/exception/RelationshipNoEntityType.java
package entities.exception;

import com.onyx.persistence.IManagedEntity;
import com.onyx.persistence.annotations.*;
import com.onyx.persistence.annotations.values.RelationshipType;

/**
 * Created by timothy.osborn on 12/14/14.
 */
@Entity
public class RelationshipNoEntityType implements IManagedEntity
{
    @Identifier
    @Attribute
    public String id = "234";

    @Relationship(type = RelationshipType.ONE_TO_MANY, inverseClass = Object.class)
    public Object relationship;
}

File: onyx-database-tests/src/main/kotlin/entities/exception/RelationshipNoEntityType.kt
package entities.exception

import com.onyx.persistence.IManagedEntity
import com.onyx.persistence.annotations.*
import com.onyx.persistence.annotations.values.RelationshipType

/**
 * Created by timothy.osborn on 12/14/14.
 */
@Entity
class RelationshipNoEntityType : IManagedEntity {

    override var referenceId: Long = 0L

    @Identifier
    @Attribute
    var id = "234"

    @Relationship(type = RelationshipType.ONE_TO_MANY, inverseClass = Any::class)
    var relationship: Any? = null
}
